### PR TITLE
Add Player.onRotateItem event

### PR DESCRIPTION
**Pull request by marmichalski**
### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Move rotating item to lua.

**Issues addressed:** #3976

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: data/scripts/eventcallbacks/player/default_onRotateItem.lua [hunk 1]
```diff
@@ -0,0 +1,10 @@
+local ec = EventCallback
+
+ec.onRotateItem = function(self, item)
+	local newId = item:getType():getRotateTo()
+	if newId ~= 0 then
```
Comment (marmichalski) [inline data/scripts/eventcallbacks/player/default_onRotateItem.lua:5]: This was and still is basically impossible, as item.isRotatable checks if rotateTo is not 0. I have just kept it as-it-was in C++ 🙈

**Comment by MillhioreBT** [comment]
You are using an outdated datapack apparently.

**Comment by marmichalski** [comment]
> You are using an outdated datapack apparently.

I am targeting 1.4 branch.

**Review by EPuncker — APPROVED** [review]

**Comment by ranisalt** [comment]
Would rather merge into master and then backport into 1.4 through cherry-picking instead. Otherwise, it's a spotless patch :star2:

**Comment by marmichalski** [comment]
> Would rather merge into master and then backport into 1.4 through cherry-picking instead. Otherwise, it's a spotless patch 🌟

This would not work as-is in master. So does not really matter where would it be merged first, it still would require patching for the other branch. 🎁

**Pull request closed without merging**